### PR TITLE
Update headline features for Foreman 3.1

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: _includes/manuals/3.1/1.2_release_notes.md [hunk 1]
```diff
@@ -1,9 +1,34 @@
 ## Release notes for {{page.version}}
 
-This section will be updated prior to the next release.
-
 ### Headline features
 
+#### Improvements and additions to the new Host Details page:
```
Comment (upadhyeammit) [inline _includes/manuals/3.1/1.2_release_notes.md:5]: @amirfefer if you think I should add or remove something then request to comment.

**Comment by upadhyeammit** [comment]
@tbrisker @evgeni @ezr-ondrej @melcorr if you think I should remove something from headline features or add some then request to comment and review.

**Comment by evgeni** [comment]
I have nothing to add/remove.

**Comment by ezr-ondrej** [comment]
I believe we are supporting Puppet 7 now, don't we?
At maintainers meeting, we were disscussing that this was the last piece in the puzzle for Puppet 7 support https://github.com/theforeman/foreman/pull/8899

**Comment by upadhyeammit** [comment]
> I believe we are supporting Puppet 7 now, don't we? At maintainers meeting, we were disscussing that this was the last piece in the puzzle for Puppet 7 support [theforeman/foreman#8899](https://github.com/theforeman/foreman/pull/8899)

I have added Puppet 7 change in snippets. Accordingly cherry picked the https://github.com/theforeman/foreman/pull/8899 in core as well

File: _includes/manuals/3.1/1.2_release_notes.md [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+The new Host Details page is no longer experimental, although it is still not the default one. One can change `Settings` -> `New hosts details UI` to Yes to make it the default option. There are few new tabs added on this page,
+
+* ##### Insights tab
+The recommendations are available for the hosts in the Insights tab when using RHEL with a Red Hat cloud subscription.
+It is feasible to begin remediation for the suggested remedies (if any are available).
+
+* ##### Ansible tab
+It has the ability to manage Ansible roles and variables.
+The recurring jobs for the host can also be scheduled from the Ansible tab.
+New PatternFly 4 search bar:
```
Comment (melcorr) [inline _includes/manuals/3.1/1.2_release_notes.md]:
```suggestion

```

**Review by melcorr — CHANGES_REQUESTED** [review]
Something small

File: _includes/manuals/3.1/1.2_release_notes.md [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
+
+* ##### Ansible tab
+It has the ability to manage Ansible roles and variables.
+The recurring jobs for the host can also be scheduled from the Ansible tab.
+
+* ##### PatternFly 4 Search tab
+A contemporary search experience is provided by the new search bar. It's the beginning of an eventual transition to PatternFly 4 for many UI components in the future.
+
+* ##### New Build button and card for BMC Power Management
+New modals for build button and BMC Power Management is available which improves exprience of new Host Details page.
```
Comment (melcorr) [inline _includes/manuals/3.1/1.2_release_notes.md]:
```suggestion
New modals for a build button and BMC Power Management is available, which improves exprience of new Host Details page.
```

File: _includes/manuals/3.1/1.2_release_notes.md [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
+
+#### Descriptions for all of the provisioning templates:
+Over a hundred new descriptions have been added to provisioning templates, snippets, and scripts to assist users understand particular usage and use them in an effective manner.
+
+#### The purge:puppet task:
+There is addition of rake task foreman-rake purge:puppet to purge puppet data if Puppet ENC functionality is not required.
```
Comment (melcorr) [inline _includes/manuals/3.1/1.2_release_notes.md]:
```suggestion
A new foreman-rake purge:puppet task has been added to purge Puppet data if Puppet ENC functionality is not required.
```

**Review by melcorr — CHANGES_REQUESTED** [review]
Some small comments but otherwise good!

**Review by ezr-ondrej — APPROVED** [review]

**Review by melcorr — APPROVED** [review]
Thank you !

**Comment by upadhyeammit** [comment]
@melcorr @ezr-ondrej thanks for review! Merging this now.

**Pull request merged**